src/org/KeyManager.groovy

```diff
@@ -0,0 +1,84 @@
+#!groovy
+
+package org
+
+class KeyManager {
+    KeyManager(String privKeyPath, String pubKeyPath, FileManager fileManager) {
+        this.privKeyPath = privKeyPath
+        this.pubKeyPath = pubKeyPath
+        this.fileManager = fileManager
+    }
+
+    public void CopyKeysTo(String destination) {
+        String pubKeyLocation = fileManager.GetFileLocation(this.pubKeyPath)
+        String privKeyLocation = fileManager.GetFileLocation(this.privKeyPath)
+        String pubKeyName = fileManager.GetFileName(this.pubKeyPath)
+        String privKeyName = fileManager.GetFileName(this.privKeyPath)
+        fileManager.Copy(pubKeyName, pubKeyName, pubKeyLocation, destination)
+        fileManager.Copy(privKeyName, privKeyName, privKeyLocation, destination)
+        this.pubKeyPath = destination + "/" + pubKeyName
+        this.privKeyPath = destination + "/" + privKeyName
+    }
+
+    public void RenamePubKey(String name) {
+        String oldName = fileManager.GetFileName(this.pubKeyPath)
+        String location = fileManager.GetFileLocation(this.pubKeyPath)
+        fileManager.Move(oldName, name, location, location)
+        this.pubKeyPath = location + "/" + name
+    }
+
+    public void RenamePrivKey(String name) {
+        String oldName = fileManager.GetFileName(this.privKeyPath)
+        String location = fileManager.GetFileLocation(this.privKeyPath)
+        fileManager.Move(oldName, name, location, location)
+        this.privKeyPath = location + "/" + name
+    }
+
+    public void AddPermissions() {
+        "chmod 600 ${this.privKeyPath}".execute()
+        "chmod 600 ${this.pubKeyPath}".execute()
+    }
+
+    public boolean PrepareKeyFiles(String machineDir) {
+        if (!this.KeysExist())
+            return false
+        this.CopyKeysTo(machineDir)
+        this.RenamePubKey("id_rsa.pub")
+        this.RenamePrivKey("id_rsa")
+        this.AddPermissions()
+        return true
+    }
+
+    public boolean KeysExist() {
+        return this.privKeyPath != "" && this.pubKeyPath != ""
+    }
+
+    // TODO: Set terraform script to not require key file when executing
+    // anything else than create method.
+    public void CreateEmptyKeyFiles(String nameForPrivKey, String nameForPubKey, String destination) {
+        fileManager.NewFile("${destination}/${nameForPrivKey}")
+        fileManager.NewFile("${destination}/${nameForPubKey}")
+        "chmod 600 ${destination}/${nameForPrivKey}".execute()
```

Comment:
execute() doesn't send output to jenkins log. We could use trim() to get the result and send it back to jenkins but it would be troublesome and in case of terraform we would have to wait till the method is done (which is like 15-50 minutes) instead of showing current state of process.
For chmod we do not need to see the result because if there was any error then other methods would fail.